test/setCommand_test.js

```diff
@@ -0,0 +1,155 @@
+goog.module('datalayerhelper.helper.testing.setCommand');
+goog.setTestOnly();
+
+const {DataLayerHelper} = goog.require('helper');
+
+describe('The set command', () => {
+  let dataLayer;
+  let dlh;
+  let commandAPI;
+  beforeAll(() => {
+    dataLayer = [];
+    commandAPI = function() {
+      dataLayer.push(arguments);
+    };
+    dlh = new DataLayerHelper(dataLayer);
+    commandAPI('set', 'bar', 'foo');
+  });
+
+  describe('with 2 argument format', () => {
+    it('updates model without corrupting it.', () =>
+    {
+      const targetModel = Object.assign({}, dlh.abstractModelInterface_);
```

Comment:
I suppose this is fine if you're concerned about the model being corrupted.

However, it looks to me like this line is repeated in every test. Maybe make targetModel a variable and set it in a "beforeEach"?

Personally, I would *also* reset the DataLayerHelper state before each test. If you need a test to rely on a particular state before executing it, you should really set up that state within the test itself.